HighscoreWindow.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2022, Harshit Sharma <harshits908@gmail.com>
+ * Copyright 2023, Johan Wagenheim
+ * This file is distributed under the terms of the MIT license
+ */
+
+#include "HighscoreWindow.h"
+#include "WindowBoard.h"
+#include "Game.h"
+
+#include <Catalog.h>
+#include <string>
+#include <LayoutBuilder.h>
+#include <Message.h>
+#include <Messenger.h>
+#include <Button.h>
+#include <String.h>
+#include <StringView.h>
+
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "HighscoreWindow"
+
+HighscoreWindow::HighscoreWindow(const char* oldHighscorer, const int32 oldHighscore, 
+	const int32 newHighscore, const BMessenger &messenger)
+	:
+	BWindow(BRect(200, 200, 1,1), 0, B_MODAL_WINDOW, B_NOT_CLOSABLE | B_NOT_RESIZABLE),
+				fMessenger(messenger)
+{
+	BStringView* congratulations = new BStringView("congratulations",
+											B_TRANSLATE("Congratulations!"));
+	congratulations->SetFont(be_bold_font);
+	congratulations->SetFontSize(25);
+	congratulations->SetAlignment(B_ALIGN_HORIZONTAL_CENTER);
+	
+	BString newHighscoreText = B_TRANSLATE("New high-score: %newhighscore%");
+	newHighscoreText.ReplaceFirst("%newhighscore%", std::to_string(newHighscore).c_str());
+	BStringView* newHighScore = new BStringView("newHighScore", newHighscoreText.String());
+	newHighScore->SetFont(be_bold_font);
+	newHighScore->SetFontSize(20);
+	newHighScore->SetAlignment(B_ALIGN_HORIZONTAL_CENTER);
+	
+	BStringView* inputlabel = new BStringView("inputlabel",
+										B_TRANSLATE("Please enter your name:"));
+	inputlabel->SetAlignment(B_ALIGN_HORIZONTAL_CENTER);
+	inputlabel->SetFont(be_bold_font);
+
+	fInputBox = new
+		BTextControl("NameInput", NULL, NULL, new BMessage(H2048_NAME_ENTERED));
+	fInputBox->SetText(B_TRANSLATE("Anonymous"));
+	fInputBox->SetAlignment(B_ALIGN_HORIZONTAL_CENTER,B_ALIGN_HORIZONTAL_CENTER);
+	fInputBox->MakeFocus();
```

Comment:
If you move this line to the very end of this constructor, the text in the text control actually gets selected and the user can start typing.